app/(main)/league/[leagueId]/entry/[entryId]/week/Week.tsx

```diff
@@ -61,6 +66,32 @@ const Week = ({ entry, league, NFLTeams, week }: IWeekProps): JSX.Element => {
     }),
   });
 
+  /**
+   * Fetches the league's weekly pick results for the user and set the user pick.
+   * @returns {Promise<void>}
+   */
+  const getUserWeeklyPick = async (): Promise<void> => {
+    try {
+      const userWeeklyPickResults = await getAllWeeklyPicks({
+        leagueId: league,
+        weekId: week,
+      });
+
+      updateWeeklyPicks({
+        leagueId: league,
+        gameWeekId: week,
+        userResults: userWeeklyPickResults || {},
+      });
+
+      if (userWeeklyPickResults?.[user.id]?.[entry]) {
+        const userPick = userWeeklyPickResults[user.id][entry].teamName;
+        setUserPick(userPick);
+      }
+    } catch (error) {
+      console.error('Error getting weekly pick:', error);
```

Comment:
I believe we want to `throw` new Error here so an error boundary can catch this. What are your thoughts on this?

Reply:
Should throw error. When making an API call and the user expects something to change, they should know that it failed. Otherwise, they'll think they actually made the pick and wonder what went wrong.